Support python3

grafi-tt opened the pull request:
This patch make editorconfig support both python2 and python3.

It seems Python3 support has been considered as tedious.
https://github.com/editorconfig/editorconfig/issues/197
By separating .py file, however, not so messy implementation is possible.

grafi-tt commented:
I think the cause of Travis's error is not my patch. It fails to install vim-gtk and I can't investigate this.

xuhdev commented:
Thanks for the PR---it looks like a great solution! I'll look into it.

xuhdev commented on plugin/editorconfig.vim:
```diff
@@ -232,47 +234,17 @@ function! s:InitializePythonBuiltin(editorconfig_core_py_dir) " {{{2
+        let s:py_eval = 'pyeval'
+    elseif has ('python3')
+        execute 'py3file ' . s:pyscript_path
+        let s:py_eval = 'py3eval'
```
Do you think it would also be a good idea to assign `pyfile` or `py3file` to a variable for future use (future modification, for example).

grafi-tt replied:
Though I'm not sure whether it can be better or useful, I think it is okay. Anyway this change is easy and I suppose there is no problem.

grafi-tt replied:
That is done on ca17e97, because it looks cleaner when the code is changed to support vim 7.3.

xuhdev commented:
Sorry for the delay, since this is a relatively big (and important) patch, I really want to be careful. I've updated the tests---your patch works well on python 2 case. However, I noticed a few issues for python 3, please see my coming comments.

xuhdev commented on plugin/editorconfig.py:
```diff
@@ -0,0 +1,42 @@
+            ec_data['options'] = editorconfig.get_properties(ec_data['filename'])
+        except editorconfig_except.EditorConfigError as e:
+            if int(vim.eval('g:EditorConfig_verbose')) != 0:
+                print >> sys.stderr, str(e)
```
I don't think this is legal in python 3...

grafi-tt replied:
Fixed on f3bf442 and c51ae80.

xuhdev commented on plugin/editorconfig.py:
```diff
@@ -0,0 +1,42 @@
+
+    except:
+        vim.command('let l:ret = 3')
+        raise
```
I noticed you added a "raise" here. However, if an exception occurs, "del sys.path[0]" would not execute. Can you fix it please?

grafi-tt replied:
Fixed on 401a948.

xuhdev commented:
It would be awesome if you can rebase the PR to master  thus the tests would work :)

grafi-tt commented:
Thank you for your comments! I rebased the PR and fixed issues.

It turned out `pyeval` function is unavailable on vim 7.3. So I changed it to `py` command. This fix is required for comparability, though `py3` command can't be used on vim 7.3 at all.

xuhdev commented:
@grafi-tt I've squashed all the commits and commited in c2b7a104b826b7ff9283d32cb95a039ddccde79b. Thanks for your contribution!

grafi-tt commented:
Thanks you!

milani commented:
So does the plugin now support `+python3` on vim? If so, should README reflect that?

xuhdev commented:
@milani  Updated. Thanks!

milani commented:
@xuhdev Thanks.

I tried the new changes. Still does not work on my vim. I have the plugin installed using Vundle on Ubuntu 18.04, vim 8.0. How can I help to find the possible problem?

xuhdev commented:
@milani You should open up a new issue and describe the details of your issue.
